### ISSUE TITLE
Clicking on a running application in the dock activates the application start animation

### ISSUE THREAD

**Issue by TriVoxel** [helloSystem/hello#343]
Every time I click on a running application's icon, it plays the "zoom and fade" animation for starting applications. While this looks fun, it certainly doesn't make a whole lot of sense. This happens when you have two or more applications open and you toggle between them using the dock. It also happens when an application is minimized and you click it's icon in the dock to un-minimize it.

**Comment by probonopd** [comment]
You are right. The Dock should only show the animation when the application is not running yet. So, we should only show the animation if we are not showing a black or white dot under the icon.

Do you think you could send a pull request?
